pkg/types/intoto/v1/intoto.proto

```diff
@@ -0,0 +1,59 @@
+//
+// Copyright 2022 The Sigstore Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+syntax = "proto3";
+package dev.sigstore.rekor.types.intoto.v1;
+
+import "google/api/field_behavior.proto";
+import "rekor_common.proto";
+
+option go_package = "github.com/sigstore/rekor/pkg/generated/protobuf/types/intoto/v1";
+option java_package = "dev.sigstore.rekor.types.intoto.v1";
+option java_multiple_files = true;
+option java_outer_classname = "IntotoV1Proto";
+
+/*
+ * ProposedIntoto represents the structure of the message required to present to Rekor
+ * to make an entry of the Intoto type.
+ */
+message ProposedIntoto {
+    /*
+     * DSSE envelope encompassing the entire Intoto statement
+     */
+    bytes envelope                                 = 1 [(google.api.field_behavior) = REQUIRED];
+    /*
+     * Public key used to verify signature(s) over envelope
+     */
+    dev.sigstore.rekor.common.PublicKey public_key = 2 [(google.api.field_behavior) = REQUIRED];
+}
+
+/*
+ * Intoto represents the structure of the entry persisted in the transparency log for
+ * the Intoto type.
+ */
+message Intoto {
```

Comment:
My biggest issue with this is that you can't take an `Intoto` entry and "verify" it the way that Rekor will verify it.

Can we add the `repeated Signatures signature` in here too? So that clients can verify the entry signature using the Signatures signing the payload_hash?